doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,250 @@
+                }
+            ],
+            "actions": [
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "-1"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Test Cases
+
+### Show FAN Status Test
+
+Show FAN status test verifies that all FAN related information can be shown correctly via `show platform fanstatus`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink FAN related sysfs and fill mock data for "presence", "speed", "status", "target_speed", "led status".
+3. Issue command `show platform fanstatus`.
+4. Record the command output.
+5. Verify that command output matches the mock data.
+6. Restore FAN related soft links in sysfs.
+
+### Show Thermal Status Test
+
+Show thermal status test verifies that all thermal related information can be shown correctly via `show platform temperature`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink thermal related sysfs and fill mock data for "temperature", "high_threshold", "high_critical_threshold".
+3. Issue command `show platform temperature`.
+4. Record the command output.
+5. Verify that command output matches the mock data.
+6. Restore FAN related soft links in sysfs.
+
+### FAN Test
+
+FAN test verifies that proper action should be taken for conditions including: FAN absence, FAN over speed, FAN under speed.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy valid_policy.json to pmon docker and backup the original one.
+3. Restart pmon service by command `systemctl restart pmon` to trigger thermal control daemon reload policy configuration file.
+4. Unlink FAN related sysfs and make mock data: first FAN absence.
+5. Wait for at least 65 seconds. Verify target speed of all FANs are set to 100% according to valid_policy.json. Verify there is a warning log for FAN absence.
+6. Make mock data: first FAN presence.
+7. Wait for at least 65 seconds. Verify target speed of all FANs are set to 60% according to valid_policy.json. Verify there is a notice log for FAN presence.
+8. Make mock data: first FAN speed exceed threshold(speed < target speed), second FAN speed exceed threshold(speed > target speed).
+9. Wait for at least 65 seconds. Verify led turns to red for first and second FAN. Verify there is a warning log for over speed and a warning speed for under speed.
+10. Make mock data: first and second FAN speed recover to normal.
+11. Wait for at least 65 seconds. Verify led turns to green for first and second FAN. Verify there are two notice logs for speed recovery.
+12. Restore FAN related soft links in sysfs. Restore the original policy file.
+
+### PSU Absence Test
+
+PSU absence test verifies that once any PSU absence, all FAN speed will be set to proper value according to policy file.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy valid_policy.json to pmon docker and backup the original one.
+3. Restart pmon service to trigger thermal control daemon reload policy configuration file.
+4. Turn off two PSUs.
```

Comment:
this one you are using the PDU function, right? no simulation?

Reply:
how can you turn off 2 PSUs? the system is not functional with no PSUs?
you probably need to shut 1, test, get it back, shut 2, test, get it back, test

Reply:
> this one you are using the PDU function, right? no simulation?

Yes, I plan to reuse the psu_controller in platform test package.

Reply:
> how can you turn off 2 PSUs? the system is not functional with no PSUs?
> you probably need to shut 1, test, get it back, shut 2, test, get it back, test

You are right. We can't shutdown two PSUs at the same time.